Create ui-c8y-1023-19-3-ui-data-point-graph-does-not-merge-y-axis-units-correctly.md (#4099)
SoftwareAG/c8y-docs: content/change-logs/application-enablement/ui-c8y-1023-19-3-ui-data-point-graph-does-not-merge-y-axis-units-correctly.md

---
date: ""

> title: UI data point graph now correctly merges y-axis units

Reviewer: ```suggestion
title: Data point graph and data explorer now correctly merge y-axis units
```

product_area: Application enablement & solutions
change_type:
  - value: change-VSkj2iV9m
    label: Fix
component:
  - value: component-YbYJ3gLU_
    label: Web SDK
build_artifact:
  - value: tc-pjJiURv9Y
    label: ui-c8y
ticket: MTM-65593
version: 1023.19.3
---

> Data graph widget and the Data Explorer previously had an issue where matching y-axis units were not merged correctly by default, causing multiple axes with the same unit to appear or show irrelevant duplicate data. This made graphs harder to read and interpret when several datapoints shared the same measurement units. The behavior has now been fixed so that duplicate units are automatically merged, improving clarity. In addition, axes with the same minimum and maximum values are also merged to reduce visual clutter. As a result, graphs now display multiple datapoints with shared units more cleanly and consistently, making them easier to read and compare.

Reviewer: ```suggestion
The "Data graph" widget and the data explorer previously had an issue where matching y-axis units were not merged correctly by default, causing multiple axes with the same unit to show irrelevant duplicate data. This made graphs harder to read and interpret when several datapoints shared the same measurement units. The behavior has now been fixed so that duplicate units are automatically merged, improving clarity. In addition, axes with the same minimum and maximum values are also merged to reduce visual clutter. As a result, graphs now display multiple datapoints with shared units more cleanly and consistently, making them easier to read and compare.
```

> Data graphs in the Data Explorer previously defaulted the y-axis to always include 0, which could make visualizations less readable when the data range did not span zero. This behavior was different from the old widget and sometimes required manual adjustments to get a meaningful view. The default y-axis behavior has now been changed so it no longer automatically starts at 0, aligning it with the legacy widget. Users can still choose to force the y-axis to start at 0 using the available option. As a result, charts and graphs now scale more appropriately by default while retaining the flexibility to include 0 when needed. This improves readability for most datasets without requiring manual configuration.

Reviewer 2: ```suggestion
Data graph widget and the Data Explorer previously had an issue where matching y-axis units were not merged correctly by default, causing multiple axes with the same unit to appear or show irrelevant duplicate data. This made graphs harder to read and interpret when several datapoints shared the same measurement units. The behavior has now been fixed so that duplicate units are automatically merged, improving clarity. In addition, axes with the same minimum and maximum values are also merged to reduce visual clutter. As a result, graphs now display multiple datapoints with shared units more cleanly and consistently, making them easier to read and compare.
```
